src/mobu/services/business/notebookrunner.py

```diff
@@ -158,6 +170,12 @@ async def execute_code(self, session: JupyterLabSession) -> None:
             self.logger.info(msg)
 
             for cell in self.read_notebook(self._notebook):
```

Comment:
I think there's an argument that we should only refresh the repository between notebooks, not between cells. If the notebook does some cleanup, well, we're going to miss it sometimes anyway, but it seems better to run it to completion before intentionally resetting, and I don't see a downside.